cli/command/stack/deploy.go

```diff
@@ -179,18 +183,20 @@ func getConfigFile(filename string) (*composetypes.ConfigFile, error) {
 func convertNetworks(
 	namespace namespace,
 	networks map[string]composetypes.NetworkConfig,
-) map[string]types.NetworkCreate {
+) (map[string]types.NetworkCreate, []string) {
 	if networks == nil {
 		networks = make(map[string]composetypes.NetworkConfig)
 	}
 
 	// TODO: only add default network if it's used
 	networks["default"] = composetypes.NetworkConfig{}
 
+	externalNetworks := []string{}
 	result := make(map[string]types.NetworkCreate)
 
 	for internalName, network := range networks {
 		if network.External.Name != "" {
+			externalNetworks = append(externalNetworks, network.External.Name)
```

Comment:
I didn't notice this before but this check `network.Externalk.Name != ""` is wrong. I think it should be `if network.External.External`.